### PR TITLE
Tickets/dm 47203

### PR DESCRIPTION
**Pull request by teweitsai**
* Migrate the reusable functions from **ts_m2gui**.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+requires = [ "setuptools", "setuptools_scm" ]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ts_guitool"
+description = """
+This package is the Python QT tool code."""
+license = { text = "GPL" }
+classifiers = [ "Programming Language :: Python :: 3" ]
+urls = {documentation = "https://jira.lsstcorp.org/secure/Dashboard.jspa", repository = "https://github.com/lsst-ts/ts_guitool" }
```
Comment (couger01) [inline pyproject.toml]:
```suggestion
urls = {documentation = "https://rubinobs.atlassian.net/secure/Dashboard.jspa", repository = "https://github.com/lsst-ts/ts_guitool" }
```

File: doc/manifest.yaml
Comment (couger01) [inline doc/manifest.yaml:1]: Please remove this file as it is not needed.

File: SConstruct
Comment (couger01) [inline SConstruct:1]: Please remove this file as it is not needed since we do not use scons for testing.

File: ups/ts_guitool.table [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# List EUPS dependencies of this package here.
+# - Any package whose API is used directly should be listed explicitly.
+# - Common third-party packages can be assumed to be recursively included by
+#   the "sconsUtils" package.
+setupRequired(sconsUtils)
```
Comment (couger01) [inline ups/ts_guitool.table]: This line is not required.

File: ups/ts_guitool.cfg
Comment (couger01) [inline ups/ts_guitool.cfg:1]: We don't use scons for building this package and so this is not needed.

**Review by couger01 — CHANGES_REQUESTED** [review]
Overall LGTM, but I have a couple of suggestions and recommendations.

File: pyproject.toml (hunk 1 above)
Comment (teweitsai) [reply pyproject.toml]: Updated in the commit: [c2950b7](https://github.com/lsst-ts/ts_guitool/pull/1/commits/c2950b7e377381a298a81fd240ce595db475b8d9). Thanks!

File: SConstruct
Comment (teweitsai) [reply SConstruct:1]: Removed in the commit: [c2950b7](https://github.com/lsst-ts/ts_guitool/pull/1/commits/c2950b7e377381a298a81fd240ce595db475b8d9). Thanks!

File: doc/manifest.yaml
Comment (teweitsai) [reply doc/manifest.yaml:1]: Removed in the commit: [c2950b7](https://github.com/lsst-ts/ts_guitool/pull/1/commits/c2950b7e377381a298a81fd240ce595db475b8d9). Thanks!

File: ups/ts_guitool.table (hunk 2 above)
Comment (teweitsai) [reply ups/ts_guitool.table]: Updated in the commit: [c2950b7](https://github.com/lsst-ts/ts_guitool/pull/1/commits/c2950b7e377381a298a81fd240ce595db475b8d9). Thanks!

File: ups/ts_guitool.cfg
Comment (teweitsai) [reply ups/ts_guitool.cfg:1]: Removed in the commit: [c2950b7](https://github.com/lsst-ts/ts_guitool/pull/1/commits/c2950b7e377381a298a81fd240ce595db475b8d9). Thanks!

**Review by couger01 — APPROVED** [review]
LGTM.

**Pull request merged**